xtext/org.icyphy.linguafranca/src/org/icyphy/generator/TypeScriptGenerator.xtend

```diff
@@ -326,6 +362,29 @@ class TypeScriptGenerator extends GeneratorBase {
     override String getTargetReference(Parameter param) {
         return '''this.«param.name».get()'''
     }
+    
+    /** Open a non-Lingua Franca import file at the specified URI
+     *  in the specified resource set. Throw an exception if the
+     *  file import is not supported. This class imports .proto files
+     *  and runs, if possible, the protoc protocol buffer code generator
+     *  to produce the required .d.ts and .js files.
+     *  @param importStatement The original import statement (used for error reporting).
+     *  @param resourceSet The resource set in which to find the file.
+     *  @param resolvedURI The URI to import.
+     */
+    override openForeignImport(Import importStatement, ResourceSet resourceSet, URI resolvedURI) {
```

Comment:
This might change in the future if we use `files` in target properties to import .proto files. Nonetheless, this is backwards compatible with the current import mechanism.